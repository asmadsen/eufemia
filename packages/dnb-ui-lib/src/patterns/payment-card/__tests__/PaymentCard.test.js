/**
 * Component Test
 *
 */

import React from 'react'
import {
  mount,
  // fakeProps,
  axeComponent,
  toJson,
  loadScss
} from '../../../core/jest/jestSetup'
import Component from '../PaymentCard'

// const defaultProps = fakeProps(require.resolve('../PaymentCard'), {
//   optional: true
// })
const defaultProps = {
  product_code: 'DNB',
  card_number: '************1337'
}

describe('PaymentCard component with urls', () => {
  const Comp = mount(<Component {...defaultProps} />)

  it('have to match snapshot', () => {
    expect(toJson(Comp)).toMatchSnapshot()
  })

  it.skip('has ...', () => {
    expect(
      Comp.find('li.dnb-payment-card--active')
        .instance()
        .getAttribute('aria-current')
    ).toBe('x')
  })

  it('should validate with ARIA rules', async () => {
    expect(await axeComponent(Comp)).toHaveNoViolations()
  })
})

describe('PaymentCard scss', () => {
  it('have to match snapshot', () => {
    const scss = loadScss(
      require.resolve('../style/dnb-payment-card.scss')
    )
    expect(scss).toMatchSnapshot()
  })
  it('have to match default theme snapshot', () => {
    const scss = loadScss(
      require.resolve('../style/themes/dnb-payment-card-theme-ui.scss')
    )
    expect(scss).toMatchSnapshot()
  })
})
