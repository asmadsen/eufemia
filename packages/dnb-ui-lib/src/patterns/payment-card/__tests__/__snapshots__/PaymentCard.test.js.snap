// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentCard component with urls have to match snapshot 1`] = `
<PaymentCard
  card_number="************1337"
  card_status="active"
  class={null}
  className={null}
  digits={8}
  id={null}
  locale={null}
  product_code="DNB"
  rawData={null}
  skeleton={false}
  text_blocked={null}
  text_card_number={null}
  text_expired={null}
>
  <Provider
    locale={null}
  >
    <figure
      className="dnb-payment-card"
    >
      <NormalCard
        cardNumber=" **** 1337 "
        cardStatus="active"
        data={
          Object {
            "cardDesign": Object {
              "bankAxept": Object {
                "is": [Function],
              },
              "bankLogo": Object {
                "color": "#fff",
              },
              "cardStyle": "card--design-default",
              "mastercard": Object {
                "is": [Function],
              },
              "name": "Default",
              "privateBanking": Object {
                "is": [Function],
              },
              "saga": Object {
                "is": [Function],
              },
              "visa": Object {
                "color": "#fff",
              },
            },
            "cardType": Object {
              "is": [Function],
            },
            "displayName": "",
            "productCode": "DNB",
            "productName": "",
            "productType": Object {
              "is": [Function],
            },
          }
        }
        id={null}
        skeleton={false}
        translations={
          Object {
            "card_number": "************1337",
            "card_status": "active",
            "children": null,
            "class": null,
            "className": null,
            "digits": 8,
            "id": null,
            "locale": null,
            "product_code": "DNB",
            "rawData": null,
            "skeleton": false,
            "text_blocked": "Sperret",
            "text_card_number": "Kortnummer",
            "text_expired": "Utgått",
          }
        }
      >
        <div
          className="dnb-payment-card__card dnb-payment-card__card--design-default"
          id={null}
        >
          <div
            className="dnb-payment-card__card__content"
          >
            <BankLogo
              logoType={
                Object {
                  "color": "#fff",
                }
              }
            >
              <SVG
                className="dnb-payment-card__card__bank-logo"
                fill="#fff"
              >
                <svg
                  className="dnb-payment-card__card__bank-logo"
                  fill="none"
                  height="38"
                  id=""
                  viewBox="0 0 56 38"
                  width="56"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clipRule="evenodd"
                    d="M10.4176 9.23538C9.2448 8.77802 7.95573 8.54587 6.58423 8.54587H0.660099C0.296531 8.54587 0 8.83052 0 9.18288V28.9445C0 29.2955 0.296531 29.5818 0.660099 29.5818H6.58423C7.95573 29.5818 9.2448 29.3493 10.4176 28.892C11.7812 28.3613 12.94 27.5393 13.8593 26.4489C15.9504 23.9729 16.2624 20.837 16.2641 19.1892C16.2641 17.787 16.0295 14.2476 13.8593 11.6784C12.94 10.5883 11.7812 9.76639 10.4176 9.23538ZM6.58425 27.6097H2.03299V10.518H6.58425C8.25707 10.518 10.6009 10.9258 12.283 12.9171C14.0909 15.0573 14.2308 18.2592 14.2308 19.1892C14.2308 20.5761 13.9777 23.2038 12.283 25.2102C10.6009 27.2019 8.25707 27.6097 6.58425 27.6097ZM42.6933 19.4514V27.6336H48.382C53.986 27.6336 53.986 24.3838 53.986 23.3153C53.986 21.1016 52.0055 19.4514 49.1728 19.4514H42.6933ZM42.6933 10.4983H47.5031C49.202 10.4983 50.4716 10.7773 51.2746 11.3282C51.8012 11.6881 52.4978 12.3688 52.4978 14.1197C52.4978 17.2557 49.9532 17.499 49.1728 17.499H42.6933V10.4983ZM52.8289 18.3095L52.7198 18.2559L52.8119 18.178C53.5875 17.5154 54.5126 16.272 54.5126 14.1197C54.5126 12.1522 53.8139 10.6761 52.4365 9.73275C51.2696 8.93404 49.656 8.54587 47.5031 8.54587H41.3146C40.9541 8.54587 40.662 8.82952 40.6603 9.17859V28.9532C40.6603 29.3023 40.9527 29.5818 41.3129 29.5818H48.382C50.6085 29.5818 52.3918 29.1248 53.6831 28.2127C54.4323 27.6842 55.0096 27.0067 55.3986 26.1992C55.7974 25.3705 56 24.4019 56 23.3153C56 21.6613 55.2743 20.0709 53.9569 19.0231C53.6184 18.7542 53.2382 18.5139 52.8289 18.3095ZM35.9165 0V28.9885C35.9165 29.2482 35.7454 29.4728 35.4916 29.5584C35.2866 29.6276 35.066 29.4973 34.9491 29.3438L21.6854 13.1224V38H19.6524V9.19199C19.6524 8.93232 19.8232 8.70389 20.077 8.62172C20.14 8.60265 20.2064 8.59213 20.2708 8.59213C20.468 8.59213 20.6526 8.68285 20.7709 8.83832L33.8835 24.9802V0H35.9165Z"
                    fill="#fff"
                    fillRule="evenodd"
                  />
                </svg>
              </SVG>
            </BankLogo>
            <CardText
              cardNumber=" **** 1337 "
              skeleton={false}
              translations={
                Object {
                  "card_number": "************1337",
                  "card_status": "active",
                  "children": null,
                  "class": null,
                  "className": null,
                  "digits": 8,
                  "id": null,
                  "locale": null,
                  "product_code": "DNB",
                  "rawData": null,
                  "skeleton": false,
                  "text_blocked": "Sperret",
                  "text_card_number": "Kortnummer",
                  "text_expired": "Utgått",
                }
              }
            >
              <span
                className="dnb-payment-card__card__wrapper"
              >
                <P
                  bold={null}
                  className="dnb-payment-card__card__holder"
                  medium={null}
                  modifier={null}
                  small={null}
                  style_type="x-small bold"
                >
                  <ForwardRef
                    className="dnb-p  dnb-p--x-small dnb-p--bold dnb-payment-card__card__holder"
                    is="p"
                  >
                    <Elem
                      _ref={null}
                      class={null}
                      className="dnb-p  dnb-p--x-small dnb-p--bold dnb-payment-card__card__holder"
                      css={null}
                      internalClass={null}
                      is="p"
                      skeleton={null}
                      skeleton_method="font"
                    >
                      <p
                        className="dnb-p  dnb-p--x-small dnb-p--bold dnb-payment-card__card__holder"
                      >
                        Kortnummer
                      </p>
                    </Elem>
                  </ForwardRef>
                </P>
                <P
                  bold={null}
                  className="dnb-payment-card__card__numbers"
                  medium={null}
                  modifier={null}
                  small={null}
                  style_type={null}
                >
                  <ForwardRef
                    className="dnb-p dnb-payment-card__card__numbers"
                    is="p"
                  >
                    <Elem
                      _ref={null}
                      class={null}
                      className="dnb-p dnb-payment-card__card__numbers"
                      css={null}
                      internalClass={null}
                      is="p"
                      skeleton={null}
                      skeleton_method="font"
                    >
                      <p
                        className="dnb-p dnb-payment-card__card__numbers"
                      >
                         **** 1337 
                      </p>
                    </Elem>
                  </ForwardRef>
                </P>
              </span>
            </CardText>
            <ProductLogo
              cardDesign={
                Object {
                  "bankAxept": Object {
                    "is": [Function],
                  },
                  "bankLogo": Object {
                    "color": "#fff",
                  },
                  "cardStyle": "card--design-default",
                  "mastercard": Object {
                    "is": [Function],
                  },
                  "name": "Default",
                  "privateBanking": Object {
                    "is": [Function],
                  },
                  "saga": Object {
                    "is": [Function],
                  },
                  "visa": Object {
                    "color": "#fff",
                  },
                }
              }
              productType={
                Object {
                  "is": [Function],
                }
              }
            />
            <TypeLogo
              cardDesign={
                Object {
                  "bankAxept": Object {
                    "is": [Function],
                  },
                  "bankLogo": Object {
                    "color": "#fff",
                  },
                  "cardStyle": "card--design-default",
                  "mastercard": Object {
                    "is": [Function],
                  },
                  "name": "Default",
                  "privateBanking": Object {
                    "is": [Function],
                  },
                  "saga": Object {
                    "is": [Function],
                  },
                  "visa": Object {
                    "color": "#fff",
                  },
                }
              }
              cardType={
                Object {
                  "is": [Function],
                }
              }
            />
          </div>
          <StatusOverlay
            cardStatus="active"
            translations={
              Object {
                "card_number": "************1337",
                "card_status": "active",
                "children": null,
                "class": null,
                "className": null,
                "digits": 8,
                "id": null,
                "locale": null,
                "product_code": "DNB",
                "rawData": null,
                "skeleton": false,
                "text_blocked": "Sperret",
                "text_card_number": "Kortnummer",
                "text_expired": "Utgått",
              }
            }
          />
        </div>
      </NormalCard>
      <figcaption
        className="dnb-sr-only dnb-payment-card__figcaption"
      />
    </figure>
  </Provider>
</PaymentCard>
`;

exports[`PaymentCard scss have to match default theme snapshot 1`] = `
"/*
* PaymentCard theme
*
*/
"
`;

exports[`PaymentCard scss have to match snapshot 1`] = `
"/*
* DNB PaymentCard
*
*/
/**
 * This file is only used to make components independent
 * so that they can get imported individually, without the core styles
 *
 */
/*
 * Utilities
 */
/*
 * Scopes
 *
 */
/*
 * Document Reset
 *
 */
.dnb-payment-card {
  font-family: var(--font-family-default);
  font-weight: var(--font-weight-default);
  font-size: var(--font-size-small);
  font-style: normal;
  line-height: var(--line-height-basis);
  color: var(--color-black-80, #333);
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  /**
  * Ensure consistency and use the same as HTML reset -> html {...}
  * between base and code package
  */
  -moz-tab-size: 4;
  tab-size: 4;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  word-break: break-word;
  /**
  * 1. Remove repeating backgrounds in all browsers (opinionated).
  * 2. Add border box sizing in all browsers (opinionated).
  */
  /**
  * 1. Add text decoration inheritance in all browsers (opinionated).
  * 2. Add vertical alignment inheritance in all browsers (opinionated).
  */
  margin: 0;
  padding: 0; }
  .dnb-payment-card *,
  .dnb-payment-card ::before,
  .dnb-payment-card ::after {
    background-repeat: no-repeat;
    /* 1 */
    box-sizing: border-box;
    /* 2 */ }
  .dnb-payment-card ::before,
  .dnb-payment-card ::after {
    text-decoration: inherit;
    /* 1 */
    vertical-align: inherit;
    /* 2 */ }

/*
 * PaymentCard component
 *
 */
:root {
  --dnb-payment-shadow-e2: 0 3px 6px rgba(0, 0, 0, 0.16),
    0 3px 6px rgba(0, 0, 0, 0.23);
  --dnb-payment-shadow-figma: 0 8px 16px rgba(51, 51, 51, 0.08);
  --dnb-payment-bg-default: linear-gradient(
    184.55deg,
    #008484 6.31%,
    #007272 82.66%
  );
  --dnb-payment-bg-white: linear-gradient(
    184.55deg,
    #fafafa 4.63%,
    var(--color-white) 92.04%
  );
  --dnb-payment-bg-silver: linear-gradient(
    184.55deg,
    var(--color-black-20) 4.63%,
    #e8e8e8 82.87%
  );
  --dnb-payment-bg-gold: linear-gradient(
    184.55deg,
    #bfa970 4.63%,
    #d3bd83 84.57%
  );
  --dnb-payment-bg-saga: linear-gradient(
    184.55deg,
    #241e23 6.31%,
    #251d23 85.2%
  );
  --dnb-payment-bg-private: var(--color-black-80);
  --dnb-payment-bg-mc-black: var(--color-black);
  --dnb-payment-bg-business-no-visa: linear-gradient(
    184.55deg,
    #50c9de 6.31%,
    #4bbed2 82.66%
  );
  --dnb-payment-bg-business-with-visa: linear-gradient(
    184.55deg,
    #2cc48e 6.31%,
    #28b482 82.66%
  ); }

.dnb-payment-card {
  margin: 0; }
  .dnb-payment-card__figcaption {
    color: white;
    opacity: 0; }
  .dnb-payment-card__card {
    position: relative;
    width: 85.6mm;
    height: 53.98mm;
    padding: 6mm;
    border-radius: 3.48mm;
    border: 1px solid;
    border-left-color: transparent;
    border-right-color: transparent;
    border-top-color: rgba(255, 255, 255, 0.1);
    border-bottom-color: rgba(0, 0, 0, 0.2);
    color: white;
    box-shadow: var(--dnb-payment-shadow-e2);
    background: #007272;
    transition: all 225ms cubic-bezier(0, 0, 0.2, 1) 0ms;
    /* White, Youth, MyFirst */ }
    .dnb-payment-card__card--design-default {
      background: var(--dnb-payment-bg-default);
      color: var(--color-white); }
    .dnb-payment-card__card--design-white {
      background: var(--dnb-payment-bg-white);
      color: var(--color-black-80); }
    .dnb-payment-card__card--design-silver {
      background: var(--dnb-payment-bg-silver);
      color: var(--color-black-80); }
    .dnb-payment-card__card--design-gold {
      background: var(--dnb-payment-bg-gold);
      color: var(--color-white); }
    .dnb-payment-card__card--design-saga {
      background: var(--dnb-payment-bg-saga);
      color: var(--color-white); }
    .dnb-payment-card__card--design-private {
      background: var(--dnb-payment-bg-private);
      color: var(--color-white); }
    .dnb-payment-card__card--design-black {
      background: var(--dnb-payment-bg-mc-black);
      color: var(--color-white); }
    .dnb-payment-card__card--design-business-no-visa {
      background: var(--dnb-payment-bg-business-no-visa);
      color: var(--color-white); }
    .dnb-payment-card__card--design-business-with-visa {
      background: var(--dnb-payment-bg-business-with-visa);
      color: var(--color-white); }
    .dnb-payment-card__card::after {
      content: '';
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
      background-image: radial-gradient(farthest-corner at 80% 30%, #f7f6e61f 0%, #06060600 60%, #06060600 99%);
      border-radius: inherit; }
    .dnb-payment-card__card__content {
      display: flex;
      height: 100%;
      width: 100%;
      position: relative; }
    .dnb-payment-card__card__wrapper {
      position: absolute;
      bottom: 0;
      left: 0; }
    .dnb-spacing .dnb-payment-card__card .dnb-p {
      margin-bottom: 0 !important; }
    .dnb-payment-card__card__holder {
      text-transform: uppercase;
      text-align: left;
      opacity: 0.5; }
      .dnb-payment-card__card__holder.dnb-p {
        font-size: 0.625rem; }
    .dnb-payment-card__card__numbers {
      font-size: var(--font-size-medium); }
    .dnb-payment-card__card__credit-type {
      height: 51px;
      position: absolute;
      bottom: 0;
      right: 0; }
    .dnb-payment-card__card__bank-logo {
      position: absolute;
      top: 0;
      left: 0; }
    .dnb-payment-card__card__product-type {
      height: 45px;
      position: absolute;
      top: 0;
      right: 0; }
  .dnb-payment-card__blocking__overlay {
    position: absolute;
    left: 0;
    top: 0;
    right: 0;
    bottom: 0;
    border-radius: inherit;
    display: flex;
    justify-content: center;
    align-items: center;
    background: linear-gradient(184.55deg, rgba(204, 204, 204, 0.5) 4.63%, rgba(232, 232, 232, 0.5) 84.83%); }
  .dnb-payment-card__blocking__center {
    width: 77px;
    height: 77px;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    background: #333;
    color: var(--color-sand-yellow);
    border-radius: 4px; }
"
`;
